operator/pkg/helmreconciler/prune.go

```diff
@@ -84,10 +90,20 @@ var (
 		schema.GroupVersionKind{Group: "admissionregistration.k8s.io", Version: "v1", Kind: name.ValidatingWebhookConfigurationStr},
 		schema.GroupVersionKind{Group: "apiextensions.k8s.io", Version: "v1", Kind: name.CRDStr},
 	)
+	autoScalingV2GVK = schema.GroupVersionKind{Group: "autoscaling", Version: "v2", Kind: name.HPAStr}
 )
 
+// modifyPruneResourcesBasedOnK8SVersion modifies the pruning resources based on the k8s version
+func (h *HelmReconciler) modifyPruneResourcesBasedOnK8SVersion() {
+	// autoscaling v2 API is available on >=1.23
+	if kube.IsAtLeastVersion(h.kubeClient, autoscalingV2MinK8SVersion) {
+		NamespacedResources = append(NamespacedResources, autoScalingV2GVK)
```

Comment:
Couldn't we end up appending this multiple times for in-cluster operator?

Reply:
updated per above comment